### PR TITLE
Refactoring for the sake of senecajs.org needs

### PR DESCRIPTION
**Pull request by rionastokes**

### CODE REVIEW

File: maintain.js [hunk 1]
```diff
@@ -163,9 +163,12 @@ module.exports = {
...
           dataForChecks.packageName = fileContent.name
           try {
-            dataForChecks.orgName = fileContent.repository.url.split('/')[3]
+            let repo_url_rx =
+              /(git@|(git|(git\+)*https):\/\/)github.com(\/|:)[a-z]+\/[a-z|-]+(.git)*/
+            dataForChecks.orgName =
+              fileContent.repository.url.match(repo_url_rx)[5]
           } catch (error) {
-            dataForChecks.orgName = '+++illegalchar+++'
+            dataForChecks.orgName = null
```
Comment (rjrodger) [inline maintain.js]: don't use a try-catch for this - keep it for actual exceptions
regex.match returns `undefined` if no match, so just check for that

File: view_orgrepo_versions.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+const Filehound = require('filehound')
+
+async function printGitURLs() {
+  const pathPromise = Filehound.create()
```
Comment (rjrodger) [inline view_orgrepo_versions.js]: no need for a separate var here, instead:
```
const pathList = await Filehound...
```

File: view_orgrepo_versions.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    let pkg = require(path)
+    if (null != pkg.homepage) {
+      hp.push(pkg.homepage)
+      totalhp++
+    } else {
+      hp.push('no hp')
+      totalhp++
+      nohp++
+    }
+    if (null != pkg.repository) {
```
Comment (rjrodger) [inline view_orgrepo_versions.js]: use `pkg?.repository` - cleaner

**Review by rjrodger — CHANGES_REQUESTED** [review]

File: view_orgrepo_versions.js (hunk 3 above)
Comment (rionastokes) [reply view_orgrepo_versions.js]: Ah very handy operator

File: view_orgrepo_versions.js (hunk 2 above)
Comment (rionastokes) [reply view_orgrepo_versions.js]: No worries

File: maintain.js (hunk 1 above)
Comment (rionastokes) [reply maintain.js]: Grand, it's been replaced with match()?.[5] set the value to undefined if needs be without having a second check

**Pull request merged**